### PR TITLE
docs: Add guide how to debug/develop GH workflows

### PR DESCRIPTION
**Pull request by jkonecny12**
This should give at least simple overview of how to debug/develop workflows with links to find out more.

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -22,3 +22,32 @@ When editing a template, the following roughly describes what to expect:
...
+
+How to debug/develop GH workflows
+=================================
+
+The installer team has a lot invested into the [GitHub actions](https://docs.github.com/en/actions). They works great, however, debugging might be a bit tricky especially when you get into point of debugging a non pull_request trigger (others are taken from default GitHub branch) workflow running on our self-hosted runners. For these cases this guide should help you to set-up required environment on your fork.
+
+With pull_request GitHub workflow trigger
+-----------------------------------------
+
+Most of the events triggering workflows needs debugging / testing on with one exception which is `pull_request` trigger:
```
Comment (KKoukiou) [inline .github/workflows/README.md:34]: I don't understand this sentence.

Comment (jkonecny12) [reply .github/workflows/README.md:34]: I see it's mixed up. I'll fix that.

**Comment by jkonecny12** [comment]
**UPDATED:** Hopefully it's better now.

File: .github/workflows/README.md [hunk 2]
```diff
@@ -22,3 +22,32 @@ When editing a template, the following roughly describes what to expect:
 
 - Whitespace handling is a complicated affair. YAML wants it precise, while Jinja is messy.
   If you stick to the two methods above, everything stays mostly deterministic.
+
+How to debug/develop GH workflows
+=================================
+
+The installer team has a lot invested into the [GitHub actions](https://docs.github.com/en/actions). They works great, however, debugging might be a bit tricky especially when you get into point of debugging a non pull_request trigger (others are taken from default GitHub branch) workflow running on our self-hosted runners. For these cases this guide should help you to set-up required environment on your fork.
```
Comment (rvykydal) [inline .github/workflows/README.md:29]: They works -> They work

**Review by rvykydal — APPROVED** [review]
Looks good to me.

File: .github/workflows/README.md (hunk 2 above)
Comment (jkonecny12) [reply .github/workflows/README.md]: Fixed, thanks!

**Review by KKoukiou — APPROVED** [review]

**Pull request merged**